src/detection/FootprintSet.cc

```diff
@@ -333,18 +333,28 @@ FootprintSet mergeFootprintSets(FootprintSet const &lhs,      // the FootprintSe
         }
     }
 
-    FootprintSet fs(*idImage, Threshold(1), 1, false);  // detect all pixels in rhs + lhs
-                                                        /*
-                                                         * Now go through the new Footprints looking up and remembering their progenitor's IDs; we'll use
-                                                         * these IDs to merge the peaks in a moment
-                                                         *
-                                                         * We can't do this as we go through the idFinder as the IDs it returns are
-                                                         *   (lhsId + 1) | ((rhsId + 1) << nbit)
-                                                         * and, depending on the geometry, values of lhsId and/or rhsId can appear multiple times
-                                                         * (e.g. if nbit is 2, idFinder IDs 0x5 and 0x6 both contain lhsId = 0) so we get duplicates
-                                                         * of peaks.  This is not too bad, but it's a bit of a pain to make the lists unique again,
-                                                         * and we avoid this by this two-step process.
-                                                         */
+    // Ensure fields added to the input peak schemas are preserved in the output.
+    table::Schema schema;
+    if (lhsFootprints.size() > 0) {
+        schema = lhsFootprints[0]->getPeaks().getSchema();
+    } else if (rhsFootprints.size() > 0) {
+        schema = rhsFootprints[0]->getPeaks().getSchema();
```

Comment:
It'd be pretty easy to add a check to see if the schemas are equal and throw an exception if they are not, and I think it's worth doing, given that not checking leaves the door open to somebody else basically repeating all the debugging you did today.